src/legacy/core_plugins/kbn_doc_views/public/views/table/table.tsx

```diff
@@ -48,8 +48,9 @@ export function DocViewTable({
           .sort()
           .map(field => {
             const valueRaw = flattened[field];
-            const value = formatValue(valueRaw, formatted[field]);
-            const isCollapsible = typeof value === 'string' && value.length > COLLAPSE_LINE_LENGTH;
+            const value = trimAngularSpan(String(formatted[field]));
+
+            const isCollapsible = value.length > COLLAPSE_LINE_LENGTH;
```

Comment:
This is the main change, by using the value, formatted by our field formatters, without any component internal modification, it's safe to use `dangerouslySetInnerHTML` later on. And since this internal formatter isn't used anymore, lot's of helper + test code can be removed.